src/array_decoder/mod.rs

```diff
@@ -98,12 +98,12 @@ impl<T: ArrowPrimitiveType> PrimitiveArrayDecoder<T> {
             derive_present_vec(&mut self.present, parent_present, batch_size).transpose()?;
         let mut data = vec![T::Native::ZERO; batch_size];
         match present {
-            Some(present) => {
+            Some(present) if present.null_count() > 0 => {
                 self.iter.decode_spaced(data.as_mut_slice(), &present)?;
                 let array = PrimitiveArray::<T>::new(data.into(), Some(present));
                 Ok(array)
             }
-            None => {
+            _ => {
```

Comment:
Thoughts on putting the check in `derive_present_vec` to share this functionality for non-primitive arrays?

https://github.com/datafusion-contrib/orc-rust/blob/f934c8eab4915836b8655e61bec5ddd959113c50/src/array_decoder/mod.rs#L211-L226

Reply:
Good point, done.